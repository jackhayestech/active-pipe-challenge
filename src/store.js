import { createStore, applyMiddleware, compose } from 'redux';
import createRootReducer from './redux/reducers/Root.reducer';

/*
*
*/
const middlewareCompose = compose(
  applyMiddleware(),
);

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const store = createStore(
  createRootReducer(),
  composeEnhancers(middlewareCompose),
);

export default store;
