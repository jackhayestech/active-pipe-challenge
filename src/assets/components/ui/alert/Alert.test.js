import React from 'react';
import { create } from "react-test-renderer";
import { Provider } from 'react-redux';
import store from '../../../../store';
import Alert from './Alert.component';

describe("The message alert item", () => {
  test("Not to display", () => {
    const component = create(<Alert display={false} />);
    expect(component.toJSON()).toMatchSnapshot();
  });

  test("To display error message", () => {
    const component = create(
      <Provider store={store}>
        <Alert display={true} type="error" />
      </Provider>);
    expect(component.toJSON()).toMatchSnapshot();
  });
});