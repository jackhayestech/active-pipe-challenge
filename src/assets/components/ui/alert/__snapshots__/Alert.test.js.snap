// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The message alert item Not to display 1`] = `null`;

exports[`The message alert item To display error message 1`] = `
<div
  className="alert error"
>
  <div
    className="alertContent"
    onClick={[Function]}
  >
    <div
      className="errorContent"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-frown fa-w-16 "
        data-icon="frown"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 496 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M248 8C111 8 0 119 0 256s111 248 248 248 248-111 248-248S385 8 248 8zm80 168c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zm-160 0c17.7 0 32 14.3 32 32s-14.3 32-32 32-32-14.3-32-32 14.3-32 32-32zm170.2 218.2C315.8 367.4 282.9 352 248 352s-67.8 15.4-90.2 42.2c-13.5 16.3-38.1-4.2-24.6-20.5C161.7 339.6 203.6 320 248 320s86.3 19.6 114.7 53.8c13.6 16.2-11 36.7-24.5 20.4z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
      <span>
        Â Uh-oh... it looks like some things haven't loaded correctly.
      </span>
    </div>
    <div>
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-sync-alt fa-w-16 "
        data-icon="sync-alt"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M370.72 133.28C339.458 104.008 298.888 87.962 255.848 88c-77.458.068-144.328 53.178-162.791 126.85-1.344 5.363-6.122 9.15-11.651 9.15H24.103c-7.498 0-13.194-6.807-11.807-14.176C33.933 94.924 134.813 8 256 8c66.448 0 126.791 26.136 171.315 68.685L463.03 40.97C478.149 25.851 504 36.559 504 57.941V192c0 13.255-10.745 24-24 24H345.941c-21.382 0-32.09-25.851-16.971-40.971l41.75-41.749zM32 296h134.059c21.382 0 32.09 25.851 16.971 40.971l-41.75 41.75c31.262 29.273 71.835 45.319 114.876 45.28 77.418-.07 144.315-53.144 162.787-126.849 1.344-5.363 6.122-9.15 11.651-9.15h57.304c7.498 0 13.194 6.807 11.807 14.176C478.067 417.076 377.187 504 256 504c-66.448 0-126.791-26.136-171.315-68.685L48.97 471.03C33.851 486.149 8 475.441 8 454.059V320c0-13.255 10.745-24 24-24z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
    </div>
  </div>
</div>
`;
